File: src/main/java/com/ject/studytrip/pomodoro/infra/jpa/PomodoroRepositoryAdapter.java
package com.ject.studytrip.pomodoro.infra.jpa;

import com.ject.studytrip.pomodoro.domain.model.Pomodoro;
import com.ject.studytrip.pomodoro.domain.repository.PomodoroRepository;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class PomodoroRepositoryAdapter implements PomodoroRepository {
    private final PomodoroJpaRepository pomodoroJpaRepository;

    @Override
    public Pomodoro save(Pomodoro pomodoro) {
        return pomodoroJpaRepository.save(pomodoro);
    }

    @Override
    public Optional<Pomodoro> findByDailyGoalId(Long dailyGoalId) {
        return pomodoroJpaRepository.findByDailyGoalId(dailyGoalId);
    }
}

File: src/main/kotlin/com/ject/studytrip/pomodoro/infra/jpa/PomodoroRepositoryAdapter.kt
package com.ject.studytrip.pomodoro.infra.jpa

import com.ject.studytrip.pomodoro.domain.model.Pomodoro
import com.ject.studytrip.pomodoro.domain.repository.PomodoroRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
class PomodoroRepositoryAdapter(
    private val pomodoroJpaRepository: PomodoroJpaRepository,
) : PomodoroRepository {
    override fun save(pomodoro: Pomodoro): Pomodoro = pomodoroJpaRepository.save(pomodoro)

    override fun findByDailyGoalId(dailyGoalId: Long): Optional<Pomodoro> = pomodoroJpaRepository.findByDailyGoalId(dailyGoalId)
}
